pkg/sources/source_manager.go

```diff
@@ -24,7 +24,8 @@ type SourceManager struct {
 	// Max number of units to scan concurrently per source.
 	concurrentUnits int
 	// Run the sources using source unit enumeration / chunking if available.
-	useSourceUnits bool
+	// Checked at runtime to allow feature flagging.
+	useSourceUnitsFunc func() bool
```

Comment:
question: This is mainly for future proofing?

Reply:
No, it's a function so it can be called at runtime / updated dynamically, instead of only at initialization.